src/EffectiveCSharp.Analyzers/PreferDeclarationInitializersToAssignmentStatementsAnalyzer.cs

```diff
@@ -0,0 +1,656 @@
+    /// If the expression is an assignment to a field, it marks the field as initialized to avoid initialization in the declaration.
+    /// It ignores all other expressions.
+    /// </summary>
+    /// <param name="node">The expression statement to process.</param>
+    /// <param name="context">The context for the syntax node analysis.</param>
+    /// <param name="fields">A dictionary of field initialization information.</param>
+    /// <param name="nodeStack">A stack to hold expression statements for further processing.</param>
+    private static void ProcessExpressionStatementFromMethod(ExpressionStatementSyntax node, SyntaxNodeAnalysisContext context, Dictionary<string, FieldInitializationInfo> fields, Stack<ExpressionStatementSyntax> nodeStack)
+    {
+        if (node.Expression is InvocationExpressionSyntax invocationExpressionSyntax)
+        {
+            if (TryGetMethodDeclaration(invocationExpressionSyntax, context, out MethodDeclarationSyntax? nestedMethodDeclaration))
+            {
+                // We found an invocation expression to a method declaration, so we push the expression statements in the method to the stack.
+                PushExpressionStatements(nestedMethodDeclaration!, nodeStack);
+            }
+        }
+        else if (node.Expression is AssignmentExpressionSyntax { Left: IdentifierNameSyntax identifierNameSyntax })
+        {
+            // We found a field assignment inside this method, so we assume the field is being initialized,
+            // and we should not initialize in the declaration.
+            string fieldName = identifierNameSyntax.Identifier.Text;
+            if (!fields.TryGetValue(fieldName, out FieldInitializationInfo fieldInfo))
+            {
+                fieldInfo = new FieldInitializationInfo(fieldName);
+                fields.Add(fieldName, fieldInfo);
+            }
+
+            fieldInfo.ShouldNotInitializeInDeclaration = true;
+        }
+    }
+
+    /// <summary>
+    /// Reports diagnostics on field declarations.
+    /// </summary>
+    /// <param name="context">The analysis context.</param>
+    /// <param name="classDeclarationSyntax">The class declaration node to find the fields in.</param>
+    /// <param name="fieldInitializationInfos">The initalization info for all fields present in a constructor.</param>
+    private static void ReportDiagnosticsOnFieldDeclarations(SyntaxNodeAnalysisContext context, ClassDeclarationSyntax classDeclarationSyntax, Dictionary<string, FieldInitializationInfo> fieldInitializationInfos)
+    {
+        foreach (FieldDeclarationSyntax field in classDeclarationSyntax.ChildNodes().OfType<FieldDeclarationSyntax>())
+        {
+            SeparatedSyntaxList<VariableDeclaratorSyntax> variables = field.Declaration.Variables;
+
+            if (variables.Count != 1)
+            {
+                // We only support single variable declarations for now.
```

Comment:
Zuh?

Reply:
Yes so apparently this is a valid scenario:

private int x, y;.

I did not even want to deal with it so we just don't handle fields that are in the same declaration. If you consider this to be worth tackling in the V1 then I can work towards getting it working